app/uk/gov/hmrc/taxhistory/model/nps/HIPNpsTaxAccount.scala

```diff
@@ -0,0 +1,195 @@
+    val (npsDescription, typeCode) = typeAndDescription.split("[(]") match {
+      case Array(desc, code) => (Some(desc.trim), code.substring(0, code.indexOf(")")).toIntOption)
+      case _                 => (None, None)
+    }
+    for {
+      amount       <- (js \ "adjustedAmount").validate[BigDecimal]
+      sourceAmount <- (js \ "sourceAmount").validateOpt[BigDecimal]
+    } yield AllowanceOrDeduction(
+      `type` = typeCode.getOrElse(0),
+      npsDescription = npsDescription.getOrElse(""),
+      amount = amount,
+      sourceAmount = sourceAmount
+    )
+  }
+  implicit val writer: OWrites[AllowanceOrDeduction]              = Json.writes[AllowanceOrDeduction]
+  //TODO: to be removed
+  def toTaAllowance(allowance: AllowanceOrDeduction): TaAllowance =
+    TaAllowance(allowance.`type`, allowance.npsDescription, allowance.amount, allowance.sourceAmount)
+  def toTaDeduction(deduction: AllowanceOrDeduction): TaDeduction =
+    TaDeduction(deduction.`type`, deduction.npsDescription, deduction.amount, deduction.sourceAmount)
+}
+
+case class HIPNpsIncomeSource(
+  employmentId: Int,
+  employmentType: Option[Int],
+  actualPUPCodedInCYPlusOneTaxYear: Option[BigDecimal],
+  deductions: List[AllowanceOrDeduction],
+  allowances: List[AllowanceOrDeduction],
+  taxCode: Option[String],
+  basisOperation: Option[Int],
+  employmentTaxDistrictNumber: Option[Int],
+  employmentPayeRef: Option[String]
+) {
+  def toIncomeSource: Option[IncomeSource] =
+    for {
+      etype   <- employmentType
+      code    <- taxCode
+      distNum <- employmentTaxDistrictNumber
+      ref     <- employmentPayeRef
+    } yield IncomeSource(
+      employmentId = this.employmentId,
+      employmentType = etype,
+      actualPUPCodedInCYPlusOneTaxYear = this.actualPUPCodedInCYPlusOneTaxYear,
+      deductions = this.deductions.map(AllowanceOrDeduction.toTaDeduction),
+      allowances = this.allowances.map(AllowanceOrDeduction.toTaAllowance),
+      taxCode = code,
+      basisOperation = this.basisOperation,
+      employmentTaxDistrictNumber = distNum,
+      employmentPayeRef = ref
+    )
+}
+
+object HIPNpsIncomeSource {
+  implicit val reader: Reads[HIPNpsIncomeSource]                                 = (js: JsValue) => {
+    val employerReference                                = (js \ "employerReference").validate[String].getOrElse("")
+    val (employmentTaxDistrictNumber, employmentPayeRef) = employerReference.split("/") match {
+      case Array(taxDistrict, payeRef) => (taxDistrict.toIntOption, Some(payeRef))
+      case _                           => (None, None)
+    }
+    //TODO: what if the string doesnt contain /
+    val employmentType                                   = (js \ "employmentRecordType").validateOpt[String].getOrElse("") match {
+      case Some("PRIMARY")   => Some(1)
+      case Some("SECONDARY") => Some(2)
+      case _                 => None
+    }
```

Comment:
This is safer as:

val employmentType = (js \ "employmentRecordType").asOpt[String] match {
  case Some("PRIMARY")   => Some(1)
  case Some("SECONDARY") => Some(2)
  case _                 => None
}